File: problem17/Solution.java
class Solution {
    public static int longestSubarray(int[] nums) {
        int prev = 0, curr = 0, max = 0;
		for (int num : nums) {
			if (num == 1)
				++curr;
			else {
				max = Math.max(max, prev + curr);
				prev = curr;
				curr = 0;
			}
		}
		max = Math.max(max, prev + curr);
		return max == nums.length ? max - 1 : max;
    }
	public static void main(String[] args) {
		int nums[] = {1,1,1};
		System.out.println(longestSubarray(nums));
	}
}

File: problem17/Solution.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        start = end = len(nums) - 1
        zeros = 0
        while end >= 0:
            if nums[end] == 0:
                zeros += 1
            end -= 1
            if zeros > 1:
                if nums[start] == 0:
                    zeros -= 1
                start -= 1
        return start - end - 1
